conftest.py

```diff
@@ -149,92 +145,150 @@ def _send_post(
         log.info("Get sensor methods")
-        return make_valid_request(SensorMethod.GET_METHODS)
+        sensor_response = make_valid_request(SensorMethod.GET_METHODS)
+        result = None
+
+        if "result" in sensor_response:
+            result = SensorInfo(**sensor_response["result"])
+
+        if "error" in sensor_response:
+            result = sensor_response["error"]
+
+        return result
 
     return _get_sensor_methods
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def set_sensor_reading_interval(make_valid_request):
     def _set_sensor_reading_interval(interval: int):
         log.info("Set sensor reading interval to %d seconds", interval)
-        return make_valid_request(
+        sensor_response = make_valid_request(
             SensorMethod.SET_READING_INTERVAL, {"interval": interval}
         )
+        result = None
+
+        if "result" in sensor_response:
+            result = SensorInfo(**sensor_response["result"])
+
+        if "error" in sensor_response:
+            result = sensor_response["error"]
+
+        return result
 
     return _set_sensor_reading_interval
 
 
-@pytest.fixture(scope = "session")
+@pytest.fixture(scope="session")
 def reset_sensor_to_factory(make_valid_request, get_sensor_info):
     def _reset_sensor_to_factory():
         log.info("Send reset firmware request to sensor")
         sensor_response = make_valid_request(SensorMethod.RESET_TO_FACTORY)
-        if sensor_response != "resetting":
-            raise RuntimeError(
-                "Sensor didn't respond to factory reset properly"
+        if "result" in sensor_response:
+            if sensor_response["result"] != "resetting":
+                raise RuntimeError("Sensor didn't respond to factory reset properly")
+
+            sensor_info = wait(
+                get_sensor_info,
+                lambda x: isinstance(x, SensorInfo),
+                tries=15,
+                timeout=1,
             )
+            if not sensor_info:
+                raise RuntimeError("Sensor didn't reset to factory property")
 
-        sensor_info = wait(
-            get_sensor_info, lambda x: isinstance(x, SensorInfo), tries=15, timeout=1
-        )
-        if not sensor_info:
-            raise RuntimeError("Sensor didn't reset to factory properly")
+            return sensor_info
 
-        return sensor_info
+        if "error" in sensor_response:
+            return sensor_response["error"]
 
     return _reset_sensor_to_factory
 
 
-@pytest.fixture(scope = "session")
+@pytest.fixture(scope="session")
 def update_sensor_firmware(make_valid_request):
     def _update_sensor_firmware():
         log.info("Send firmware update request to sensor")
-        return make_valid_request(SensorMethod.UPDATE_FIRMWARE)
+        sensor_response = make_valid_request(SensorMethod.UPDATE_FIRMWARE)
+        result = None
+
+        if "result" in sensor_response:
+            result = SensorInfo(**sensor_response["result"])
```

Comment:
`update_sensor_firmware` doesn't return sensor info, so we shouldn't try to instantiate a `SensorInfo` object here. Since we don't have a class definition for what this sensor method returns, we just need to return `sensor_response["result"]`.